src/server/actions/invite.actions.ts

```diff
@@ -0,0 +1,192 @@
+                email: 'An account with this email already exists. Please sign in to accept the invitation.',
+            })
+        }
+
+        const clerkUser = await client.users.createUser({
+            emailAddress: [email],
+            password: form.password,
+            firstName: form.firstName,
+            lastName: form.lastName,
+            // publicMetadata for roles will be set by onPendingUserLoginAction
+        })
+        return clerkUser.id // Return Clerk User ID
+    } catch (error: unknown) {
+        if (isClerkApiError(error)) {
+            const clerkError = error.errors[0]
+            logger.warn({ message: 'Clerk API error during user creation', clerkError, email })
+            throw new ActionFailure({ form: clerkError.longMessage || clerkError.message })
+        }
+        logger.error({ message: 'Non-Clerk error during user creation', error, email })
+        throw new ActionFailure({ form: 'Could not create account due to an unexpected error.' })
+    }
+}, CreateAccountSchema)
+
+const ClaimInviteSchema = z.object({
+    inviteId: z.string(),
+})
+
+export const claimInviteAction = userAction(async ({ inviteId }) => {
+    const { user: authUser } = await actionContext()
+
+    const siUserId = authUser.id!
+    const { clerkId: clerkUserId } = await db
+        .selectFrom('user')
+        .select('clerkId')
+        .where('id', '=', siUserId)
+        .executeTakeFirstOrThrow()
+
+    const pendingUser = await db
+        .selectFrom('pendingUser')
+        .innerJoin('org', 'org.id', 'pendingUser.orgId')
+        .select([
+            'pendingUser.id as pendingUserId',
+            'pendingUser.email',
+            'pendingUser.orgId as localOrgDbId',
+            'pendingUser.isResearcher',
+            'pendingUser.isReviewer',
+            'org.slug as orgSlug',
+            'org.name as orgName',
+        ])
+        .where('pendingUser.id', '=', inviteId)
+        .where('pendingUser.claimedByUserId', 'is', null)
+        .executeTakeFirst()
+
+    if (!pendingUser) {
+        return { success: false, error: 'Invalid or already claimed invitation.' }
+    }
+
+    if (pendingUser.email !== authUser.primaryEmailAddress?.emailAddress) {
+        return { success: false, error: 'This invitation is for a different user. Please log out and try again.' }
+    }
+
+    const clerkOrg = await findClerkOrganization({
+        slug: pendingUser.orgSlug,
+    })
+
+    // 1. Update local DB
+    const orgMembershipDetails = await findOrCreateOrgMembership({
+        userId: siUserId,
+        slug: pendingUser.orgSlug,
+        isResearcher: pendingUser.isResearcher,
+        isReviewer: pendingUser.isReviewer,
+        isAdmin: false, // Default for invitees
+    })
+
+    try {
+        const client = await clerkClient()
+        // 2. Add to Clerk Organization (or update role if already member)
+        try {
+            await client.organizations.createOrganizationMembership({
```

Comment:
`findOrCreateOrgMembership` does this

Reply:
yes, but for now we are ignoring the case of a user possibly being in multiple orgs.  We do not want to spend any effort past what we have right now on them, because we are _likely_ to stop using Clerk orgs, as well as the clerk org switcher  altogether very soon once the permissions work progresses further